### PR TITLE
Add value format to Settings

### LINKED ISSUE #382: Consider Settings refactoring
As Karl mentioned before, we're already have two issues (#372 and #356 ) that shows us that the current structure of the Setting object does not suit us well.
We cannot change type of setting stored if we need some kind of serialization, etc, etc.

So - why don't we create some kind of custom settings object not in Entity-attribute-value paradigm, but with classic relation DB approach, where one column will be equal one or more settings (maybe via store_accessor, stored in json instead of text... but it's unnecessary details atm), and one table entry will be equal the whole settings set for application?

We might leave Setting class as a settings wrapper for some kind of backward compatibility as well.

@karlerikounapuu , @artur-beljajev your thoughts?

### PR DESCRIPTION
**Pull request by karlerikounapuu**
Closes #382 

- Every setting is now required to have `value_format` which defaults to `string`. Available formats are `string`, `boolean`, `integer`, `hash`, ` array`

- Settings are no longer accessed via Setting.setting_name because it has too many redundant methods.

- Each invidual setting must be retrieved via `Setting.find_by(code: :setting_code).retrieve`. It has same output as previous querying via class method.

`data_migrations:set_correct_setting_formats` rake task must be ran to set correct format for already existing settings.

### CODE REVIEW

File: app/models/setting.rb [hunk 1]
```diff
@@ -3,107 +3,66 @@ class Setting < ApplicationRecord
...
-    Setting.find_by(code: :registration_term).value.to_i
+  def boolean_format
+    value == 'true'
   end
 
-  def self.auction_duration
-    value = Setting.find_by(code: :auction_duration).value
+  def hash_format
+    return JSON.parse(value)[I18n.locale.to_s] if code == 'terms_and_conditions_link'
+    return JSON.parse(value)[I18n.locale.to_s] if code == 'violations_count_regulations_link'
```
Comment (yulgolem) [inline app/models/setting.rb]: How about to make a separate method for this?
Something like `localised_value`? We are returning the same value of hash in two different conditions - maybe we could join them somehow?

File: app/models/setting.rb [hunk 2]
```diff
@@ -3,107 +3,66 @@ class Setting < ApplicationRecord
...
 
-    Setting.find_by(code: :auction_duration).value.to_i
+    JSON.parse(value)
+  end
+
+  def array_format
+    JSON.parse(value)
```
Comment (yulgolem) [inline app/models/setting.rb]: .to_a? Maybe we should enforce array as a return here?

Comment (karlerikounapuu) [reply app/models/setting.rb]: Doesn't really change current behaviour, but sure, for better clarity I'm willing to add it 🙂

File: app/models/setting.rb (hunk 1 above)
Comment (karlerikounapuu) [reply app/models/setting.rb]: Might be even better. Thanks for suggestion.

**Pull request merged**